### ISSUE TITLE
Can no longer shallow copy `Kokkos::UnorderedMap` on Device

### ISSUE THREAD

**Issue by stanmoore1** [kokkos/kokkos#6066]
With latest Kokkos `develop` branch we can no longer shallow copy `Kokkos::UnorderedMap` on device. It fails with:

```
error: reference to __host__ function 'UnorderedMap' in __host__ __device__ function
```

**Comment by stanmoore1** [comment]
Note: works with v4.0 and below, but was broken sometime after 4.0 in `develop`.

**Comment by masterleinad** [comment]
From `slack`, the code in question looks like
```C++
   auto d_map_hash = k_map_hash.const_view<DeviceType>();
```

**Comment by ldh4** [comment]
Git bisect shows that this is caused by #5983 where a `shared_ptr` was introduced into `UnorderedMap`.

**Comment by masterleinad** [comment]
> Git bisect shows that this is caused by #5983 where a `shared_ptr` was introduced into `UnorderedMap`.

Right, that's what I suspected.

**Comment by ldh4** [comment]
I think we should consider changing that `shared_ptr` to a host view.

**Comment by masterleinad** [comment]
Fixed by #6073.
